lms/djangoapps/edraak_forus/tests/tests.py

```diff
@@ -0,0 +1,388 @@
+            username='The_Best_ForUs_User',
+            password='random_password',
+            honor_code=True,
+        ))
+        self.assertTrue(mock_set_logged_in_cookies.called, 'Login cookies was not set!')
+        self._assertLoggedIn(msg_prefix='The user is not logged in after clicking a course')
+
+        self.client.logout()
+        self.client.session.clear()
+
+        self._assertLoggedOut()
+
+        res_auth_2 = self.client.get(self.auth_url, self._build_forus_params())
+
+        self.assertIsInstance(res_auth_2, HttpResponseRedirect)
+        self.assertTrue(
+            expr=res_auth_2['Location'].endswith(self.dashboard_url),
+            msg='Auth does not redirect dashboard. It redirects to: `{}`'.format(res_auth_2['Location']),
+        )
+
+        self._assertLoggedIn(msg_prefix='The user is not logged in after clicking the form another time')
+
+    def test_custom_registration_messages(self):
+        res = self.client.get(self.auth_url, self._build_forus_params())
+        self.assertNotContains(res, 'Create a new account')
+        self.assertContains(res, 'Edraak account using ForUs')
+
+        self.assertNotContains(res, 'Create your account')
+        self.assertContains(res, 'Create your Edraak account')
+
+        self.assertContains(res, 'toggle-form hidden')
+
+    def _build_forus_params(self, **kwargs):
+        params = build_forus_params(course_id=unicode(self.course.id), email=self.user_email, forus_hmac='dummy_hmac')
+        params.update(**kwargs)
+        return params
+
+
+class ForUsMessagePageTest(TestCase):
+    """
+    Test message page
+    """
+    def setUp(self):
+        super(ForUsMessagePageTest, self).setUp()
+
+        self.url = reverse('forus_v1_message')
+
+    def test_message_page_with_no_error(self):
+        """
+        We don't want to the word "error" to appear in the message page.
+        """
+
+        message = 'The course was not found.'
+
+        res = self.client.get(self.url, {
+            'message': message,
+        })
+
+        self.assertContains(res, message, msg_prefix='The message is missing from the page')
+        self.assertNotContains(res, 'error', msg_prefix='The page contains the work `error` which is confusing')
+
+    def test_no_xss(self):
+        message = '<script>alert("Hello")</script>'
+        escaped_message = html_escape(message)
+
+        self.assertNotEqual(message, escaped_message, 'Something is wrong, message is not being escaped!')
+        self.assertNotIn('<script>', escaped_message, 'Something is wrong, message is not being escaped!')
+
+        res = self.client.get(self.url, {
+            'message': message,
+        })
+
+        self.assertNotContains(res, message, msg_prefix='The page is XSS vulnerable')
+        self.assertContains(res, escaped_message, msg_prefix='The page encodes the message incorrectly')
+
+
+@ddt.ddt
+@override_settings(LANGUAGE_CODE='en')
```

Comment:
That's bad, I think we're having a bug in the view/middleware in terms of language handling. I've never liked what I've done there, this has revealed a possible issue.

Let's study what edX does and try to do find alternative implementation.